### PR TITLE
improve the doc building optionally allowing signature in the docstring

### PR DESCRIPTION
**Pull request by CarloLucibello**
Also it fixes an unreported issue, where functions extend from Base, such as `join` and `intersect`, both docstrings from Base and LightGraphs are included in the docs.

### CODE REVIEW

File: doc/generators.md [hunk 1]
```diff
@@ -1,11 +1,10 @@
 <!-- AUTOGENERATED. See 'doc/build.jl' for source. -->
 ## Random Graphs
 *LightGraphs.jl* implements three common random graph generators:
-### erdos_renyi
 ```
-erdos_renyi(n::Integer, ne::Integer)
-erdos_renyi(n::Integer, p::Real)
+erdos_renyi(n::Integer, p::Real; is_directed=false, seed::Integer=-1)
 ```
```
Comment (sbromberger) [inline doc/generators.md]: ...I think this is a problem. We have two very different methods for `erdos_renyi()` - one takes two Integers, and the other takes an int and a real. The PR will remove very useful information about one of the more common methods of creating an E-R graph.

**Comment by CarloLucibello** [comment]
travis fail is unrelated. This is ready to merge

**Comment by sbromberger** [comment]
Hold up a sec - let's be perfectly clear about what this PR does. It not only fixes docs (thanks, btw!) but it's changing the signature of erdos_renyi. Can you make that a separate change?

I don't really like bundling up multiple disparate fixes in small PRs - it's ok for the sweeping refactorization we occasionally do, but this seems wrong.

**Comment by CarloLucibello** [comment]
I removed the change in erdos_renyi signature, This should be ready now

File: src/randgraphs.jl [hunk 2]
```diff
@@ -37,7 +41,7 @@ Note also that Erdős–Rényi graphs may be generated quickly using `erdos_reny
 or the  `Graph(nv, ne)` constructor, which randomly select `ne` edges among all the potential
 edges.
 """
-function erdos_renyi(n::Integer, p::Real; is_directed=false, seed::Integer=-1)
+function erdos_renyi(n::Integer, p::AbstractFloat; is_directed=false, seed::Integer=-1)
```
Comment (sbromberger) [inline src/randgraphs.jl]: What's the purpose of this change? (Why wouldn't we allow Rationals here?)

Comment (CarloLucibello) [inline src/randgraphs.jl]: I wanted to enforce `p` to be a floating point number, and for example an integer, for which we have `Integer<:Reals`. But since I hadn't considered Rational type and since integers are already captured by the other signature of erdos_renyi I'm reverting this change (I didn't even remember doing that, the change I was considering before was is_directed -> directed)

File: src/randgraphs.jl [hunk 3]
```diff
@@ -38,6 +42,7 @@ or the  `Graph(nv, ne)` constructor, which randomly select `ne` edges among all
 edges.
 """
 function erdos_renyi(n::Integer, p::Real; is_directed=false, seed::Integer=-1)
+    @assert 0 <= p <= 1
```
Comment (sbromberger) [inline src/randgraphs.jl]: 1) This should be a more informative error to the user, I think.
2) We should not be changing functionality in a PR that is supposed to be about docstring enhancements.

**Comment by CarloLucibello** [comment]
is there something else withstanding this?
